lib/api/core/validation/index.js

```diff
@@ -1,690 +1,680 @@
-  var
-    promises = [],
-    specification = {};
+    return getValidationConfiguration(this.kuzzle)
+      .then(validation => {
+        this.rawConfiguration = validation;
 
-  return getValidationConfiguration(this.kuzzle)
-    .then(validation => {
-      this.rawConfiguration = validation;
+        Object.keys(this.rawConfiguration).forEach(indexName => {
+          Object.keys(this.rawConfiguration[indexName]).forEach(collectionName => {
+            const promise = new Bluebird(resolve => {
+              this.curateCollectionSpecification(indexName, collectionName, this.rawConfiguration[indexName][collectionName])
+                .then(curatedSpec => {
+                  if (!specification.hasOwnProperty(indexName)) {
+                    specification[indexName] = {};
+                  }
 
-      Object.keys(this.rawConfiguration).forEach(indexName => {
-        Object.keys(this.rawConfiguration[indexName]).forEach(collectionName => {
-          var promise = new Promise((resolve) => {
-            this.curateCollectionSpecification(indexName, collectionName, this.rawConfiguration[indexName][collectionName])
-              .then(curatedSpec => {
-                if (!specification.hasOwnProperty(indexName)) {
-                  specification[indexName] = {};
-                }
+                  specification[indexName][collectionName] = curatedSpec;
 
-                specification[indexName][collectionName] = curatedSpec;
+                  this.kuzzle.pluginsManager.trigger('log:info', `Validation specification for ${indexName} / ${collectionName} has been loaded.`);
 
-                this.kuzzle.pluginsManager.trigger('log:info', `Validation specification for ${indexName} / ${collectionName} has been loaded.`);
+                  resolve({});
 
-                resolve({});
+                  return null;
+                })
+                .catch(error => {
+                  this.kuzzle.pluginsManager.trigger('log:error', `Specification for the collection ${collectionName} triggered an error`);
+                  this.kuzzle.pluginsManager.trigger('log:error', `Error: ${error.message}`);
 
-                return null;
-              })
-              .catch(error => {
-                this.kuzzle.pluginsManager.trigger('log:error', `Specification for the collection ${collectionName} triggered an error`);
-                this.kuzzle.pluginsManager.trigger('log:error', `Error: ${error.message}`);
+                  resolve({});
 
-                resolve({});
+                  return null;
+                });
+            });
 
-                return null;
-              });
+            promises.push(promise);
           });
-
-          promises.push(promise);
         });
+
+        return Bluebird
+          .all(promises)
+          .then(() => {
+            this.specification = specification;
+            return this.kuzzle.pluginsManager.trigger('log:info', 'All validation specifications have been treated.');
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 118 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)